### PR TITLE
JEST test suit

### PR DESCRIPTION
**Pull request by SevlaMare**
### write functions and test for:
- [x] capitalize(string)
- [x] reverseString(string)
- [x] calculator with add, subtract, divide and multiply.
- [x] Caesar Cipher
- [x] Array Analysis with average, min, max and length.

### after review
- [x] remove  Webpack
- [x] reorganize files
- [x] capitalize as a named function
- [x] reverseString as a named function
- [x] calculator as an explicit object
- [x] array analysis as an explicit object
- [x] cypher should be a single function, -x to decrypt +x to encrypt 
- [x] no longer encrypt punctuation

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Start @team :clap: However, some requirements need to be implemented in order to get approval for this milestone. Check :arrow_down: 

#### Implementation (required) :warning: 

#### `capitalize` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/capitalize_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

* [x] `capitalize` is a function that takes a string as an argument and returns that string with the first character capitalized
* [x] There is a Jest test for the `capitalize` function and it's passing

#### `reverseString` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/reverse_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

* [x] `reverseString` is a function that takes a string as an argument and returns that string reversed
* [x] There is a Jest test for the `reverseString` function and it's passing

#### `calculator` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/calculator_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

* [x] It is an object
* [x] It has an `add` method
* [x] It has a `subtract`
* [x] It has a `multiply` method
* [x] It has a `divide` method
* [x] There is a Jest test for the `calculator` object that tests the `add`, `subtract`, `multiply` and `divide` methods

#### Caesar Cipher ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/caesar_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

* [x] Caesar Cipher is implemented
* [x] It has a test for wrapping from `z` to `a`
* [x] It has a test for keeping the same case
* [x] It has a test for punctuation

#### Array Analysis ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/array_analysis_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

* [x] It has a function that takes an array of numbers as an argument and returns an object
* [x] The returned object has the `average` of the numbers in the array as a property
* [x] The returned object has the `min` of the numbers in the array as a property
* [x] The returned object has the `max` of the numbers in the array as a property
* [x] The returned object has the `length` of the numbers in the array as a property
* [x] There is a Jest test for function to calculate correctly `average`, `min`, `max` and `length`

### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by wathigo — CHANGES_REQUESTED** [review]
## Changes Required.
Great work so far on the project :clap: 
Let's make s few more changes before we move on.

### REQUIRED.
**Caesar Cipher**
* Encodes the received string using a key between 0 and 25.
After calling `cipher` method with the following params:
  ```ruby
  caesarCipher.cipher("attack at dawn",5)
  ```
  We get the following output:
  ```ruby
  "fyyfhp%fy%ifbs"
  ```
  This is the expected output:
  ```ruby
  "fyyfhp fy ifbs"
  ```
*  Keeps the original case.
We expect an output of: 
  ```ruby
    "FYYFHP FY IFBS"
  ```
  After calling the method as:
  ```ruby
    caesarCipher.cipher("ATTACK AT DAWN",5)
   ```
  But we receive:
  ```ruby
    "&99&(0%&9%)&<3"
  ```
* Works with strings that use . as separator.
Calling:
```ruby
caesarCipher.cipher("ATTACK.AT.DAWN",5)
``
Returns:
```ruby
"&99&(03&93)&<3"
```
Instead of:
```ruby
"FYYFHP.FY.IFBS"
```

**After making the changes, submit another request for review.**

**Happy Hacking :computer: 
Simon Wathigo.**

File: test/caesarCipher.test.js [hunk 1]
```diff
@@ -9,11 +9,27 @@ describe('should encrypt and decrypt correctly', () => {
...
     expect(CesarCipher.cipher('"/-@', -1)).toBe('!.,?');
   });
+
+  it('should encrypt correctly with spaces', () => {
+    expect(CesarCipher.cipher('attack at dawn', 5)).toBe('fyyfhp fy ifbs');
+  });
+
+  it('should encrypt correctly with spaces and uppercase', () => {
+    expect(CesarCipher.cipher('ATTACK AT DAWN', 5)).toBe('FYYFHP FY IFBS');
+  });
```
Comment (shloch) [inline test/caesarCipher.test.js]: good

**Review by shloch — APPROVED** [review]
Student brought along the requested changes
Really professional and responsive

### Nice job @SevlaMare , @AlfredoElizarraras 
Screen Shot 2020-09-09 at 9 47 46 PM

### Happy coding

**Pull request merged**